### PR TITLE
New Provider: Freshbooks

### PR DESCRIPTION
**Pull request by bengoshow**
Adding Freshbooks OAuth 2 authentication

## Reasoning 💡

Adding Freshbooks OAuth 2 authentication

## Checklist 🧢

- [x] Documentation
~- [ ] Tests~
- [x] Ready to be merged

This can be reviewed and merged.

### CODE REVIEW

File: types/providers.d.ts [hunk 1]
```diff
@@ -72,6 +72,7 @@ export type OAuthProviderType =
   | "FACEIT"
   | "FortyTwo"
   | "Foursquare"
+  | "Freshbooks"
```
Comment (lluia) [inline types/providers.d.ts:75]: Can we also test the types for this new provider [here](https://github.com/nextauthjs/next-auth/blob/a72a5d78a9a66f65c8fe59d4054f2ad45eb491da/types/tests/providers.test.ts#L259)? 😊

File: www/docs/providers/freshbooks.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+title: Freshbooks
+---
+
+## Documentation
+
+https://www.freshbooks.com/api/authenticating-with-oauth-2-0-on-the-new-freshbooks-api
+
+## Configuration
+
+https://my.freshbooks.com/#/developer
```
Comment (lluia) [inline www/docs/providers/freshbooks.md:12]: Can we add an extra section pointing to the options supported by this provider?
```md
## Options

The Freshbooks Provider comes with a set of default options:

- [Freshbooks Provider options](link/to/options/documentation)

You can override any of the options to suit your own use case.
```

**Review by 0ubbe — CHANGES_REQUESTED** [review]
Thanks 💚  , few changes before we can get this in 👍🏽

**Review by 0ubbe — APPROVED** [review]
Thanks 👍🏽

**Pull request merged**